.travis.yml

```diff
@@ -36,4 +36,6 @@ install:
   - python -m pip install .
 script:
   - make -f Makefile
+  - # Debug...
+  - which python
```

Comment:
`type -a python` is infinitely better than `which python` - it prints out *all* locations where `python` appears in the `PATH` (not just the first one) and it detects stuff like shell aliases and functions.

Reply:
Oh, and you may want to do `ls -l "$(type -p python)"` to check whether the file that `python` points to is a symlink. If the symlink points into a system directory, macOS will still treat the binary as protected, even if the symlink itself is not in a system directory.

(`type -p python` does basically the same thing as `which python`, except it handles some weird edge cases better due to `type` being a Bash builtin.)

Reply:
Looks like the problem here may be caused by pyenv (and is something [they're aware of, but know they can't fix](https://github.com/pyenv/pyenv-virtualenv/issues/112)). pyenv is installed in /usr/local

What is a mystery to me is why it *does* work if you use the subprocess.run() trick...

Reply:
I'm going to guess it's got something to do with the fact subprocess creates a subshell... 

At this point, I've spent *way* more time on futzing with CI than I ever intended :-) The `subprocess` hack isn't pretty, but it works. I've clarified the comment to be a little less misleading (hopefully), but otherwise, I'll leave it at that.

Reply:
OK, so the issue is that `pyenv`'s shims call out to `/usr/local/.../pyenv`, which is considered protected, so `DYLD_LIBRARY_PATH` is lost at that point.

I think this could be solved by using `DYLD_LIBRARY_PATH=whatever "$(pyenv which python)" setup.py test`. This asks `pyenv` to give us the path to the *real* `python` binary (not the shim) that is located under `~/.pyenv`, which we then call directly. That way there are no "protected" processes in the call chain.

> What is a mystery to me is why it _does_ work if you use the subprocess.run() trick...

This is likely because `pyenv exec` implicitly adds the selected Python version's real location at the front of the `PATH`:

https://github.com/pyenv/pyenv/blob/4e0ba2f47d97acd1284439cff57af059ce376b9d/libexec/pyenv-exec#L5-L14

With the new `PATH`, `python` will resolve to the real Python binary and not a shim, which will work for the same reason - no "protected" binary is called along the way.

Reply:
If you want to try this solution, I would suggest removing the `pyenv global`/`pyenv local` call, so that the shims are never generated (we should have no need for them anymore with the new solution, and this way we can't call them by accident).

Reply:
Great googly moogly... $(pyenv which python) WORKS!